src/evaluate.jl

```diff
@@ -249,17 +256,8 @@ function evaluate(a::TaylorN{T}, vals::NTuple{N,S}) where
     R = promote_type(T,S)
     a_length = length(a)
     suma = zeros(R, a_length)
-    for homPol in 1:length(a)
-        sun = zero(R)
-        for (i, a_coeff) in enumerate(a.coeffs[homPol].coeffs)
-            iszero(a_coeff) && continue
-            tmp = vals[1]^(coeff_table[homPol][i][1])
-            for n in 2:N
-                tmp *= vals[n]^(coeff_table[homPol][i][n])
-            end
-            sun += a_coeff * tmp
-        end
-        suma[homPol] = sun
+    @inbounds for homPol in length(a):-1:1
+        suma[homPol] = evaluate(a.coeffs[homPol], vals)
     end
 
     return sum( sort!(suma, by=abs2) )
```

Comment:
What is the need for the sort here?

Reply:
The short answer is to keep precision. There is no canonical Horner rule for multivariable polynomials, as far as I know, so what @blas-ko did was to save the result of each monomial evaluation in a vector, which is sorted before adding its elements. In that way, the sum is computed from the smallest number (wrt `abs2`) to the largest. This is certainly not the fastest way of doing it, but it warranties to maintain precision.